src/pages/CustomRichTextEditorLink.tsx

```diff
@@ -0,0 +1,189 @@
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { Button, ButtonVariants } from '@/ui/Button';
+import { FormElement } from '@/ui/FormElement';
+import { Inline } from '@/ui/Inline';
+import { Input } from '@/ui/Input';
+import { Stack } from '@/ui/Stack';
+
+/**
+ * Source: https://github.com/jpuri/react-draft-wysiwyg/blob/master/src/components/Option/index.js#L8
+ */
+function Option({
+  onClick,
+  children,
+  value,
+  className,
+  activeClassName = '',
+  active = false,
+  disabled = false,
+  title,
+}) {
+  return (
+    <div
+      className={classNames('rdw-option-wrapper', className, {
+        [`rdw-option-active ${activeClassName}`]: active,
+        'rdw-option-disabled': disabled,
+      })}
+      onClick={() => {
+        if (!disabled) {
+          onClick(value);
+        }
+      }}
+      aria-selected={active}
+      title={title}
+    >
+      {children}
+    </div>
+  );
+}
+
+/**
+ * Source: https://github.com/jpuri/react-draft-wysiwyg/blob/master/src/controls/Link/Component/index.js#L203
+ */
+
+function CustomRichTextEditorLink({
+  config,
+  currentState,
+  doCollapse,
+  expanded,
+  onChange,
+  onExpandEvent,
+  translations,
+}: CustomRichTextEditorLinkProps) {
+  const { t } = useTranslation();
+  const [showModal, setShowModal] = useState(false);
+  const [linkTarget, setLinkTarget] = useState('');
+  const [linkTitle, setLinkTitle] = useState('');
+  const [linkTargetOption, setLinkTargetOption] = useState(
+    config.defaultTargetOption,
+  );
+
+  useEffect(() => {
+    if (expanded && !expanded) {
```

Comment:
```suggestion
    if (expanded === false) {
```
Is this the expected behaviour? The current check looks strange